source/index.ts

```diff
@@ -185,7 +185,18 @@ is.symbol = isOfType<symbol>('symbol');
 is.numericString = (value: unknown): value is string =>
 	is.string(value) && !is.emptyStringOrWhitespace(value) && !Number.isNaN(Number(value));
 
-is.array = Array.isArray;
+is.array = <T = unknown>(value: unknown, assertion?: (value: T) => value is T): value is T[] => {
+	if (!Array.isArray(value)) {
+		return false;
+	}
+
+	if (!assertion) {
+		return true;
+	}
+
+	return value.every(assertion);
```

Comment:
```suggestion
	return Array.isArray(value) && !assertion || value.every(assertion);
```

Reply:
Hi, I know that this can be shortened to a single line, but IMO this feels less readable.

Reply:
I agree with @Arnovsky here... I had to triple-check to be convinced this was equivalent. Also the [no-mixed-operators](https://eslint.org/docs/rules/no-mixed-operators) rule would complain here by the way. I vote for keeping it as is :)